package/MDAnalysis/coordinates/GSD.py

```diff
@@ -0,0 +1,123 @@
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""GSD trajectory reader  --- :mod:`MDAnalysis.coordinates.GSD`
+============================================================
+
+Classes
+-------
+
+.. autoclass:: GSDReader
+   :inherited-members:
+
+"""
+from __future__ import absolute_import, division
+
+import numpy as np
+from . import base
+try :
+    import gsd.hoomd
+except :
+    pass
+
+class GSDReader(base.ReaderBase):
+    """Reader for the GSD format.
+
+    TODO: write docstring
+
+    """
+    format = 'GSD'
+    units = {'time': None, 'length': None}
+    _Timestep = base.Timestep
+
+    def __init__(self, filename, **kwargs):
+        """
+        Parameters
+        ----------
+        filename : str
+            trajectory filename
+        **kwargs : dict
+            General reader arguments.
+
+
+        .. versionadded:: 0.17.0
+        """
+        super(GSDReader, self).__init__(filename, **kwargs)
+        self.filename = filename
+        self.open_trajectory()
+        self.n_atoms = self._file[0].particles.N
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+        self._read_next_timestep()
+
+    def open_trajectory (self) :
+        """opens the trajectory file using gsd.hoomd module"""
+        # self._frame = -1
+        self._file = gsd.hoomd.open (self.filename,mode='rb')
+
+    def close(self):
+        """close reader"""
+        self._file.file.close()
+
+    @property
+    def n_frames(self):
+        """number of frames in trajectory"""
+        return len(self._file)
+
+    def _reopen(self):
+        """reopen trajectory"""
+        self.close()
+        self.open_trajectory()
+
+    def _read_frame (self, frame):
+        try :
+            myframe = self._file[frame]
+        except IndexError :
+            raise IOError
+
+        # sets the Timestep object
+        self.ts.frame = frame
+        self.ts.data['step'] = myframe.configuration.step
+
+        # set frame box dimensions
+        self.ts.dimensions = myframe.configuration.box
+        for i in range(3,6) :
+            self.ts.dimensions[i] = np.arccos(self.ts.dimensions[i]) * 180.0 / np.pi
+
+        # set particle positions
+        frame_positions = myframe.particles.position
+        n_atoms_now = frame_positions.shape[0]
+        if n_atoms_now != self.n_atoms :
+            raise ValueError("Frame %d has %d atoms but the initial frame has %d"
+                " atoms. MDAnalysis in unable to deal with variable"
+                " topology!"%(frame, n_atoms_now, self.n_atoms))
+        else :
+            self.ts.positions = frame_positions
+        return self.ts
+
+    def _read_next_timestep (self) :
+        """read next frame in trajectory"""
+        return self._read_frame (self.frame + 1)
+
+    @property
```

Comment:
We can't add dimensions here, it's a property of Timesteps/AtomGroups etc, but not Readers

Reply:
Okay, removed